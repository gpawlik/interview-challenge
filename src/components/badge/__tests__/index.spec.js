import * as React from "react";
import renderer from "react-test-renderer";

import { Badge } from "../";

jest.mock("../styles", () => ({
  Container: "Container",
}));

const initialProps = {
  text: "Hello World",
};

const mockComponent = (props) => {
  const finalProps = {
    ...initialProps,
    ...props,
  };

  return renderer.create(<Badge {...finalProps} />);
};

describe("Badge", () => {
  it("renders correctly default version", () => {
    const tree = mockComponent().toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with no text", () => {
    const tree = mockComponent({ text: "" }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
