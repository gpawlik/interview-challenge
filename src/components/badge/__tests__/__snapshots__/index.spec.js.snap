// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly default version 1`] = `
<Container>
  Hello World
</Container>
`;

exports[`Badge renders correctly with no text 1`] = `
<Container>
  
</Container>
`;
